dask/dataframe/io/sql.py

```diff
@@ -226,6 +226,10 @@ def _read_sql_chunk(q, uri, meta, engine_kwargs=None, **kwargs):
     engine.dispose()
     if len(df) == 0:
         return meta
+    elif len(meta.dtypes.to_dict()) == 0:
+        # only index column in loaded
+        # required only for pandas < 1.0.0
+        return df
```

Comment:
This case is only required for pandas < 1.0.0 as the following failes for these old versions:

```python
import pandas as pd
df = pd.DataFrame(index=pd.Int64Index(range(10), name='id'))
df.astype({}, copy=False)
```
Should we respect that edge case with this additional condition? Any other ideas?

Reply:
We currently require "pandas >= 0.25.0", for reference. How old is 1.0.0 now?